### PR TITLE
Let Gardener manage NATGateway when VPC ID is set

### PR DESCRIPTION
**Pull request by EmoinLanyu**
**How to categorize this PR?**

/area networking
/area documentation
/kind enhancement
/platform alicloud

**What this PR does / why we need it**:
In the past, users have to bring their own NATGateway alongside the VPC. Since AliCloud migrated to Enhanced NATGateway, a VSwitch must be created prior to the creation of the Enhanced NATGateway. The documentation for this extension was not clear about this prerequisite.

The PR enhances the documentation, and also provide an option for the users to leave the management of NATGateway to Gardener. They can now only provide a VPC. And if `networks.vpc.gardenerManagedNATGateway` is set to `true`, Gardener will create and manage the NATGateway created in the user provided VPC.

**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:
We need to test below create scenarios:
- [x] User creates a Shoot with workers in one zone, without specifying VPC id.
- [x] User creates a Shoot with workers in one zone, specifying a VPC id but not setting `gardenerManagedNATGateway`.
- [x] User creates a Shoot with workers in one zone, specifying a VPC id and setting `gardenerManagedNATGateway` to `false`.
- [x] User creates a Shoot with workers in one zone, specifying a VPC id and setting `gardenerManagedNATGateway` to `true`.
- [x] User creates a Shoot with workers in multiple zones, specifying a VPC id and setting `gardenerManagedNATGateway` to `true`, without specifying `eipAllocationID`.
- [x] User creates a Shoot with workers in multiple zones, specifying a VPC id and setting `gardenerManagedNATGateway` to `true`, and specifying `eipAllocationID`.

We also need to test upgrade scenarios:
- [x] User already has a Shoot with workers in one zone, without specifying VPC id. Reconciliation is triggered.
- [x] User already has a Shoot with workers in one zone, specifying a VPC id. Reconciliation is triggered.
- [x] User already has a Shoot with workers in multiple zones, specifying a VPC id, without specifying `eipAllocationID`.  Reconciliation is triggered.
- [x] User already has a Shoot with workers in multiple zones, specifying a VPC id, and specifying `eipAllocationID`. Reconciliation is triggered.

We can also consider testing above scenarios with default `gardenerManagedNATGateway` set to `true`.

**Release note**:

```other user
User can now let Gardener manage NATGateway in the user-provided VPC by specifying `networks.vpc.gardenerManagedNATGateway` to `true` when `networks.vpc.id` is set.
```

### CODE REVIEW

File: docs/usage-as-end-user.md [hunk 1]
```diff
@@ -120,6 +121,10 @@ The `networks.vpc` section describes whether you want to create the shoot cluste
 * If `networks.vpc.cidr` is given then you have to specify the VPC CIDR of a new VPC that will be created during shoot creation.
 You can freely choose a private CIDR range.
 * Either `networks.vpc.id` or `networks.vpc.cidr` must be present, but not both at the same time.
+* When `networks.vpc.id` is present, In addition, you can also choose to set `networks.vpc.gardenerManagedNATGateway`. It is by default `false`. When set to `true`,
```
Comment (jia-jerry) [inline docs/usage-as-end-user.md:124]:
```suggestion
* When `networks.vpc.id` is present, in addition, you can also choose to set `networks.vpc.gardenerManagedNATGateway`. It is by default `false`. When it is set to `true`,
```

**Comment by shaoyongfeng** [comment]
you may also need to change the caller logic of vpcClient.GetVPCInfo(ctx, vpcID) in actuator.go. Otherwise there will be an error popped up if no natgateway available in vpc provided by user

**Comment by shaoyongfeng** [comment]
what happens if user sets gardenerManagedNATGateway as true at first, and change it to false?  The terraform state file might become inconsistent.

File: pkg/apis/alicloud/validation/infrastructure.go [hunk 2]
```diff
@@ -103,12 +103,25 @@ func ValidateInfrastructureConfig(infra *apisalicloud.InfrastructureConfig, node
...
+func ValidateVPCChange(newVPC, oldVPC *apisalicloud.VPC, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldVPC.ID, newVPC.ID, fldPath.Child("id"))...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldVPC.CIDR, newVPC.CIDR, fldPath.Child("cidr"))...)
+
+	if oldVPC.GardenerManagedNATGateway != nil && *oldVPC.GardenerManagedNATGateway && (newVPC.GardenerManagedNATGateway == nil || !*newVPC.GardenerManagedNATGateway) {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("gardenerManagedNATGateway"), "can't change gardenerManagedNATGateway if it is set true"))
+	}
+
+	return allErrs
```
[the remark this answers is not in the document]
Comment (jia-jerry) [reply pkg/apis/alicloud/validation/infrastructure.go]: OK. I will remove it.

**Review by shaoyongfeng — APPROVED** [review]

**Pull request merged**